packages/IMAPClient-UI.package/ICFolderDialog.class/instance/update.st

```diff
@@ -3,7 +3,8 @@ update
 
 	self numberOfFetchedFolders: 0. 
 	self numberOfFolders: 
-		(ICHelper allFolders: (self emailAccounts collect: [:each | each rootFolder])) size.
+		(self emailAccounts collect: [:each | each rootFolder recursiveChildFolders]) flatten size +
```

Comment:
Instead of `collect: [...] flatten size` you could do `collect: [... size] sum`. This might be faster, but I'm not sure. If you do so, be aware that `sum` is not implemented well and does throw an error if it gets called on an empty collection

Reply:
After discussion we consented against that as the gained speed might be there but is marginal and we can not assure that `emailAccounts` is not empty